src/otx/v2/adapters/torch/lightning/anomalib/engine.py

```diff
@@ -0,0 +1,220 @@
+"""OTX adapters.torch.lightning.anomalib.Engine API."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Union
+
+import numpy as np
+from anomalib.data.inference import InferenceDataset
+from anomalib.data.utils import InputNormalizationMethod, get_transforms
+from anomalib.post_processing import NormalizationMethod, ThresholdMethod
+from anomalib.utils.callbacks import (
+    MetricsConfigurationCallback,
+    MinMaxNormalizationCallback,
+    PostProcessingConfigurationCallback,
+)
+from anomalib.utils.loggers import AnomalibTensorBoardLogger
+from pytorch_lightning.core.datamodule import LightningDataModule
+from torch.utils.data import DataLoader
+
+from otx.v2.adapters.torch.lightning.engine import LightningEngine
+
+from .registry import AnomalibRegistry
+
+if TYPE_CHECKING:  # pragma: no cover
+    import pytorch_lightning as pl
+    import torch
+    from pytorch_lightning.loggers.logger import Logger
+    from pytorch_lightning.trainer.connectors.accelerator_connector import (
+        _PRECISION_INPUT,
+    )
+    from pytorch_lightning.utilities.types import EVAL_DATALOADERS
+    from torch.optim import Optimizer
+
+PREDICT_FORMAT = Union[str, Path, np.ndarray]
+
+
+class AnomalibEngine(LightningEngine):
+    """Anomalib engine using PyTorch and PyTorch Lightning."""
+
+    def __init__(
+        self,
+        work_dir: str | Path | None = None,
+        config: str | dict | None = None,
+        task: str = "classification",
+    ) -> None:
+        """Initialize the Anomalib engine.
+
+        Args:
+            work_dir (Optional[Union[str, Path]], optional): The working directory for the engine. Defaults to None.
+            config (Optional[Union[str, dict]], optional): The configuration for the engine. Defaults to None.
+            task (str, optional): The task to perform. Defaults to "classification".
+        """
+        super().__init__(work_dir=work_dir, config=config, task=task)
+        self.registry = AnomalibRegistry()
+
+    def _update_logger(
+        self,
+        logger: list[Logger] | Logger | None = None,
+        target_path: str | None = None,
+    ) -> list[Logger] | Logger | None:
+        """Update the logger and logs them to the console or use AnomalibTensorBoardLogger.
+
+        Args:
+            logger(list[Logger] | Logger | None): Input of loggers
+            target_path(str | None): logger's target output path
+
+        Returns:
+            list[Logger] | Logger | None: Updated loggers.
+        """
+        self.trainer_config.pop("logger", None)
+        if logger is not None:
+            if isinstance(logger, list):
+                return logger
+            return [logger]
+        return [AnomalibTensorBoardLogger(save_dir=self.work_dir, name=target_path)]
```

Comment:
Can you explain why tensorboard logger is used as default?

Reply:
I've configured Anomalib to use the same logic it used in the old OTX v1. Are we sure we don't need this?

Reply:
not sure :) It is not important though since we will work on this for anomalib v1.0